connector/protobuf/src/test/resources/protobuf/functions_suite.proto

```diff
@@ -170,4 +170,41 @@ message timeStampMsg {
 message durationMsg {
   string key = 1;
   Duration duration = 2;
-}
\ No newline at end of file
+}
+
+message OneOfEvent {
```

Comment:
Are you testing more OneOf and recusion in the same message? Could you split them into separate messages?

Reply:
@rangadi I see a lot of use cases for the "payload" Oneof the field and recursive fields in it. So I thought combining Oneof with recursion would be a good test. will separate

Reply:
Combined one is fine, we could keep it. Better to have a simpler separate tests as well.